scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt

```diff
@@ -19,36 +21,32 @@ class MongoRepository(
     private val dataBase: String,
     private val mongoClient: MongoClient
 ) {
-    fun getAllSubmissions(): FindIterable<SubmissionDoc> {
-        val database = mongoClient.getDatabase(dataBase)
-        val collection = database.getCollection(SUBMISSION_COLLECTION, SubmissionDoc::class.java)
+    fun getNotImportedSubmissionsBySourceFile(sourceFile: String): FindIterable<SubmissionDoc> =
+            getSubmissionCollection().find(and(eq("sourceFile", sourceFile), eq("imported", false)))
 
-        return collection.find()
-    }
+    suspend fun save(submissionDoc: SubmissionDoc) = getSubmissionCollection().insertOne(submissionDoc)
 
-    suspend fun save(submissionDoc: SubmissionDoc) {
-        val database = mongoClient.getDatabase(dataBase)
-        val collection = database.getCollection(SUBMISSION_COLLECTION, SubmissionDoc::class.java)
-        collection.insertOne(submissionDoc)
-    }
+    suspend fun save(errorDoc: ErrorDoc) = getCollection(ERRORS_COLLECTION, ErrorDoc::class.java).insertOne(errorDoc)
 
-    suspend fun save(errorDoc: ErrorDoc) {
-        val database = mongoClient.getDatabase(dataBase)
-        val collection = database.getCollection(ERRORS_COLLECTION, ErrorDoc::class.java)
-        collection.insertOne(errorDoc)
-    }
+    suspend fun update(submissionDoc: SubmissionDoc) = getSubmissionCollection().updateOne(submissionDoc)
 
     suspend fun saveSubFile(file: File, accNo: String): ObjectId {
-        val database = mongoClient.getDatabase(dataBase)
-        val collection = database.getCollection(FILES_COLLECTION, FileDoc::class.java)
+        val collection = getFilesCollection()
+
         collection.insertOne(FileDoc(file.name, file.absolutePath, accNo))
-        return collection.findOne(Filters.eq("path", file.absolutePath))!!.id
+        return collection.findOne(eq("path", file.absolutePath))!!.id
     }
 
     suspend fun getSubFiles(ids: List<ObjectId>): List<FileDoc> {
-        val database = mongoClient.getDatabase(dataBase)
-        val collection = database.getCollection(FILES_COLLECTION, FileDoc::class.java)
+        val collection = getFilesCollection()
 
-        return ids.map { collection.findOne(Filters.eq("_id", it))!! }.toList()
+        return ids.map { collection.findOne(eq("_id", it))!! }.toList()
     }
+
+    private fun getFilesCollection() = getCollection(FILES_COLLECTION, FileDoc::class.java)
+
+    private fun getSubmissionCollection() = getCollection(SUBMISSION_COLLECTION, SubmissionDoc::class.java)
+
+    private fun <A, B : Class<A>> getCollection(name: String, documentClass: B) =
+            mongoClient.getDatabase(dataBase).getCollection(name, documentClass)
```

Comment:
this one is the perfect case for kotlin reified

```
inline fun <reified T> getCollection(name:String)
```

in that way it can be called like

getCollection<Submission/Doc>("Submissions")